designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala

```diff
@@ -40,14 +40,12 @@ trait DeploymentManager extends AutoCloseable {
     */
   def deploy(processVersion: ProcessVersion, deploymentData: DeploymentData, canonicalProcess: CanonicalProcess, savepointPath: Option[String]): Future[Option[ExternalDeploymentId]]
 
+  // TODO: add cancel for given deployment id
   def cancel(name: ProcessName, user: User): Future[Unit]
 
   def test[T](name: ProcessName, canonicalProcess: CanonicalProcess, scenarioTestData: ScenarioTestData, variableEncoder: Any => T): Future[TestResults[T]]
 
-  /**
-    * Gets status from the engine
-    */
-  def getProcessState(name: ProcessName)(implicit freshnessPolicy: DataFreshnessPolicy): Future[WithDataFreshnessStatus[Option[StatusDetails]]]
+  def getProcessStates(name: ProcessName)(implicit freshnessPolicy: DataFreshnessPolicy): Future[WithDataFreshnessStatus[List[StatusDetails]]]
```

Comment:
I wonder do we really need here a list of the states? This list is only specific for the Flink.. Maybe better will be still return Option[StatusDetails] and in the Flink implementation use `InconsistentStateDetector.extractAtMostOneStatus`? In fact, at the end we have tun invoke `extractAtMostOneStatus`.

Reply:
As we discussed offline, List will stay here